glommio/src/channels/local_channel.rs

```diff
@@ -295,37 +324,22 @@ impl<T> Drop for LocalSender<T> {
         // Will not wake up senders, but we are dropping the sender so nobody
         // wants to wake up anyway.
         state.send_waiters.take();
-
-        if let Some(ref mut waiters) = &mut state.recv_waiters {
-            let waiters = core::mem::replace(waiters, VecDeque::new());
-            drop(state);
-
-            for w in waiters {
-                w.wake();
-            }
-        }
+        wake_up_all(&mut state.recv_waiters);
     }
 }
 
 impl<T> Drop for LocalReceiver<T> {
     fn drop(&mut self) {
         let mut state = self.channel.state.borrow_mut();
         state.recv_waiters.take();
-
-        if let Some(ref mut waiters) = &mut state.send_waiters {
-            let waiters = core::mem::replace(waiters, VecDeque::new());
-            drop(state);
-
-            for w in waiters {
-                w.wake();
-            }
-        }
+        wake_up_all(&mut state.send_waiters);
     }
 }
 
 impl<T> Stream for LocalReceiver<T> {
     type Item = T;
 
+    #[inline]
```

Comment:
Is this really necessary?

Usually I find it better to leave these decisions to the compiler, and have exceptions be justified.

I am not that versed in the behaviors of the rust compiler here. So if it is notoriously bad at inlining a function like this I am happy to go for it.

Reply:
I try to only place `#[inline]` at functions/methods where I find it justified, which means that the function is:
1. really short
2. probably often called in loops or other "hot" sections
3. visible/exported outside of the crate (inside of the crate, it doesn't matter, I think the compiler maybe even ignores `#[inline]` if both caller and callee belong to the same crate, inside of the crate, only `#[inline(always)]` and `#[inline(never)]` *should* matter, I suppose).

Reply:
All those criteria are good criteria on when to inline the function. The biggest question to me is whether rust wouldn't do it itself.

All in all, I don't really mind and I merged it as is. Thanks!